extmod/modussl_axtls.c

```diff
@@ -187,12 +201,13 @@ STATIC mp_uint_t socket_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_t arg, i
 }
 
 STATIC mp_obj_t socket_setblocking(mp_obj_t self_in, mp_obj_t flag_in) {
-    // Currently supports only blocking mode
-    (void)self_in;
-    if (!mp_obj_is_true(flag_in)) {
-        mp_raise_NotImplementedError(NULL);
-    }
-    return mp_const_none;
+    mp_obj_ssl_socket_t *o = MP_OBJ_TO_PTR(self_in);
+    o->blocking = mp_obj_is_true(flag_in);
+    mp_obj_t sock = o->sock;
+    mp_obj_t dest[3];
+    mp_load_method(sock, MP_QSTR_setblocking, dest);
+    dest[2] = flag_in;
+    return mp_call_method_n_kw(1, 0, dest);
```

Comment:
I think you should first call `mp_call_method_n_kw` and only if it succeeds, then set `o->blocking`. Because if it fails at this point then the ssl socket is left in an unstable state.

Reply:
> I think you should first call `mp_call_method_n_kw` and only if it succeeds, then set `o->blocking`

Fixed.